posts/models.py

```diff
@@ -8,17 +8,21 @@
 from django.http import Http404
 
 from users.models import User
+from circles.models import Circle
 
 class Post(models.Model):
-    """Post model that stores all information of a post
+    """
+    Post model that stores all information of a post
     """
     content = models.CharField(max_length=256)
     author = models.ForeignKey(User)
     # the post of which this is a repost
-    original_post = models.ForeignKey('self', null=True, blank=True)
+    repost_original = models.ForeignKey('self', null=True, blank=True, related_name='repost')
+    reply_original = models.ForeignKey('self', null=True, blank=True, related_name='reply')
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
     favorites = models.ManyToManyField(User, related_name='favorites')
+    circles = models.ForeignKey(Circle, null=True, blank=True, related_name='circles')
```

Comment:
In case we decide us later to make able to select multiple circles. If it matters then I will change it no problem